iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -402,10 +406,18 @@ private Schema schema(Properties properties, org.apache.hadoop.hive.metastore.ap
     }
   }
 
-  private static PartitionSpec spec(Schema schema, Properties properties,
+  private static PartitionSpec spec(Configuration configuration, Schema schema, Properties properties,
       org.apache.hadoop.hive.metastore.api.Table hmsTable) {
 
-    if (hmsTable.getParameters().get(InputFormatConfig.PARTITION_SPEC) != null) {
+    if (SessionState.get().getQueryState(configuration.get(HiveConf.ConfVars.HIVEQUERYID.varname))
```

Comment:
Maybe a util class to get Icebeerg objects from the `SessionState`?

Reply:
+ 1
I'm working with the QueryState too on a different jira, and we would definitely benefit from some util methods to simplify these operations.

Reply:
I'll create a new `IcebergSessionUtil` for providing some util methods, we can sync up with @lcspinter on the implementation

Reply:
I agree. I created an util class